admin/class-h5p-plugin-admin.php

```diff
@@ -156,6 +159,79 @@ function add_settings_link($links) {
     return $links;
   }
 
+  /**
+   * Spawn an h5p content via POST request
+   */
+  public function spawn_h5p() {
+    $plugin = H5P_Plugin::get_instance();
+    $action = $_REQUEST['action'];
+    if ($action) {
+      $core = $plugin->get_h5p_instance('core'); // Make sure core is loaded
+      if ($action === 'create_h5p') {
+        // Handle creation of new h5p content.
+        $json = file_get_contents('php://input');
+
+        $decoded = json_decode($json);
+        if(!isset($decoded->library) || !isset($decoded->params)){
+          $core->h5pF->setErrorMessage(__('Invalid POST parameters set', $this->plugin_slug));
+          return;
+        }
+
+        // Construct an h5p content array with POST params
+        $content['library'] = $decoded->library;
+        $content['params'] = $decoded->params;
+
+        $content['library'] = $core->libraryFromString($content['library']);
```

Comment:
Please assign this to an intermediate variable (`$library`?) Lines 191..192 in particular should get much less hairy, and you can validate its existence before mutating the `$content` object, which should make the code slightly more readable.